csharp/extractor/Semmle.Util/EnvironmentVariables.cs

```diff
@@ -7,6 +7,8 @@ public class EnvironmentVariables
         public static string? GetExtractorOption(string name) =>
             Environment.GetEnvironmentVariable($"CODEQL_EXTRACTOR_CSHARP_OPTION_{name.ToUpper()}");
 
+        public static string? GetScratchDirectory() => Environment.GetEnvironmentVariable("CODEQL_EXTRACTOR_CSHARP_SCRATCH_DIR");
```

Comment:
Out of curiosity - when and where is this environment variable set?

Reply:
It's set in the CodeQL CLI. (I sent you a DM.)

Reply:
Yes, great!